src/main/java/me/hydos/blaze4d/mixin/integration/GlStateManagerMixin.java

```diff
@@ -220,15 +221,16 @@ private static void clearDepth(double depth, CallbackInfo ci) {
 
     @Inject(method = "_getString", at = @At("HEAD"), cancellable = true)
     private static void getString(int glStringId, CallbackInfoReturnable<String> ci) {
+        final VkPhysicalDeviceProperties properties = VkPhysicalDeviceProperties.create();
+        VK10.vkGetPhysicalDeviceProperties(Blaze4D.rosella.common.device.newDevice.getDevice().getPhysicalDevice(), properties);
         ci.setReturnValue(
                 Blaze4D.rosella == null ? "Device not initialized" :
                         switch (glStringId) {
-//                            case GL.GL_VENDOR -> tryParseVendorId(Blaze4D.rosella.common.device.properties.vendorId);
-//                            case GL.GL_EXTENSIONS -> Blaze4D.rosella.common.device.combinedExtensionsString;
-//                            case GL.GL_RENDERER -> Blaze4D.rosella.common.device.properties.deviceName;
-//                            case GL.GL_VERSION -> "Vulkan API " + Blaze4D.rosella.common.device.properties.apiVersion;
-//                            default -> throw new IllegalStateException("Unexpected value: " + glStringId);
-                            default -> "FIXME: BROKEN. use new Device system";
+                            case GL.GL_VENDOR -> tryParseVendorId(properties.vendorID());
```

Comment:
what does this look like in game?

Reply:
![image](https://user-images.githubusercontent.com/30385023/128676924-7de08318-6ac8-4524-8b02-fa363e386dea.png)

Reply:
ideally this should be fixed within rosella